components/omega/src/ocn/VertCoord.cpp

```diff
@@ -23,53 +23,101 @@ VertCoord *VertCoord::DefaultVertCoord = nullptr;
 std::map<std::string, std::unique_ptr<VertCoord>> VertCoord::AllVertCoords;
 
 //------------------------------------------------------------------------------
-// Begin initialization of default vertical coordinate, requires prior
-// initialization of Decomp.
-void VertCoord::init1() {
+// Initialize the default vertical coordinate, requires prior initialization
+// of Decomp. The optional arguments simplify the use of the VertCoord in some
+// unit tests. If ReadStream is false, the InitialVertCoord stream will not be
+// read during construction. If a value for NVertLayers is passed as argument,
+// the dimension will not be read from the mesh file.
+void VertCoord::init(
+    const bool
+        ReadStream, //< [in] optional argument to read stream, true by default
+    const int
+        InNVertLayers //< [in] optional argument to set NVertLayers explicitly
+                      //< instead of reading dimension from mesh file
+) {
 
    Decomp *DefDecomp = Decomp::getDefault();
 
-   VertCoord::DefaultVertCoord = create("Default", DefDecomp);
-
-} // end init1
-
-//------------------------------------------------------------------------------
-// Complete initialization of default vertical coordinate, requires prior
-// initialization of HorzMesh.
-void VertCoord::init2() {
-
    Config *OmegaConfig = Config::getOmegaConfig();
 
-   DefaultVertCoord->completeSetup(OmegaConfig);
+   VertCoord::DefaultVertCoord =
+       create("Default", DefDecomp, OmegaConfig, ReadStream, InNVertLayers);
 
-} // end init2
+} // end init
 
 //------------------------------------------------------------------------------
-// Construct a new VertCoord instance given a Decomp. New object is incomplete
-// and completeSetup must be called afterwards
+// Construct a new VertCoord instance given a Decomp.
 VertCoord::VertCoord(const std::string &Name_, //< [in] Name for new VertCoord
-                     const Decomp *Decomp      //< [in] associated Decomp
+                     const Decomp *Decomp,     //< [in] associated Decomp
+                     Config *Options,          //< [in] configuration options
+                     const bool ReadStream,    //< [in] logical to read stream
+                     const int InNVertLayers   //< [in] int to set vertical dim
```

Comment:
Is it valid to have `ReadStream = true` and non-zero `InVertLayers` ? If not, can check for that case and abort ?